src/Symfony/Component/DependencyInjection/Compiler/AttributeAutoconfigurationPass.php

```diff
@@ -11,40 +11,102 @@
 
 namespace Symfony\Component\DependencyInjection\Compiler;
 
+use Symfony\Component\DependencyInjection\Argument\BoundArgument;
+use Symfony\Component\DependencyInjection\Argument\ServiceLocatorArgument;
+use Symfony\Component\DependencyInjection\Argument\TaggedIteratorArgument;
+use Symfony\Component\DependencyInjection\Attribute\BindTaggedIterator;
+use Symfony\Component\DependencyInjection\Attribute\BindTaggedLocator;
 use Symfony\Component\DependencyInjection\ChildDefinition;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\LogicException;
 
 /**
  * @author Alexander M. Turek <me@derrabus.de>
  */
-final class AttributeAutoconfigurationPass implements CompilerPassInterface
+final class AttributeAutoconfigurationPass extends AbstractRecursivePass
 {
+    /** @var array<string, callable>|null */
+    private $argumentConfigurators;
+
     public function process(ContainerBuilder $container): void
     {
         if (80000 > \PHP_VERSION_ID) {
             return;
         }
 
-        $autoconfiguredAttributes = $container->getAutoconfiguredAttributes();
+        $this->argumentConfigurators = [
+            BindTaggedIterator::class => static function (BindTaggedIterator $attribute) {
+                return new TaggedIteratorArgument($attribute->tag, $attribute->indexAttribute);
+            },
+            BindTaggedLocator::class => static function (BindTaggedLocator $attribute) {
+                return new ServiceLocatorArgument(new TaggedIteratorArgument($attribute->tag, $attribute->indexAttribute));
+            },
+        ];
+
+        parent::process($container);
+
+        $this->argumentConfigurators = null;
```

Comment:
What if `parent::process()` throws an exception? 

Does it make sense to do `try/finally` just as `AbstractRecursivePass::process()` does?

Reply:
The property is gone after Nicolas' review.